alembic/config.py

```diff
@@ -200,7 +200,7 @@ def file_config(self):
         self.config_args["here"] = here
         file_config = ConfigParser(self.config_args)
         if self.config_file_name:
-            file_config.read([self.config_file_name])
+            file_config.read([self.config_file_name], "utf-8")
```

Comment:
this should be ` locale.getencoding()`

setting this to utf-8 is a possible breaking change if people use strange encodings.

Now I don't expect this to be the case, but either leave the default (so what's done now) or we should use the default format for the locale

Reply:
I don't believe that's correct. If you would like locale, my understanding is that this should be "locale" which would allow read to use the proper locale option. Yes, that resolves to that ultimately, but it ultimately requires more maintenance as you will have to keep track of how to resolve the locale instead of allowing the standard library to do so for you.

Reply:
Maybe I misunderstood https://docs.python.org/3/library/io.html#io.text_encoding though.

But my thought process is that `.read` is implemented as:

```
    def read(self, filenames, encoding=None):
        """Read and parse a filename or an iterable of filenames.

        Files that cannot be opened are silently ignored; this is
        designed so that you can specify an iterable of potential
        configuration file locations (e.g. current directory, user's
        home directory, systemwide directory), and all existing
        configuration files in the iterable will be read.  A single
        filename may also be given.

        Return list of successfully read files.
        """
        if isinstance(filenames, (str, bytes, os.PathLike)):
            filenames = [filenames]
        encoding = io.text_encoding(encoding)
        read_ok = []
        for filename in filenames:
            try:
                with open(filename, encoding=encoding) as fp:
                    self._read(fp, filename)
            except OSError:
                continue
            if isinstance(filename, os.PathLike):
                filename = os.fspath(filename)
            read_ok.append(filename)
        return read_ok
```

So `io.text_encoding` would return `locale` if passed in or `"utf-8"` / `"locale"` if `None` (currently the case). Since locale is a valid option as seen here, this is what seems to make sense for your ask.

This then goes to `open` which is discussed in https://docs.python.org/3/library/functions.html#open

> encoding is the name of the encoding used to decode or encode the file. This should only be used in text mode. The default encoding is platform dependent (whatever [locale.getencoding()](https://docs.python.org/3/library/locale.html#locale.getencoding) returns), but any [text encoding](https://docs.python.org/3/glossary.html#term-text-encoding) supported by Python can be used. See the [codecs](https://docs.python.org/3/library/codecs.html#module-codecs) module for the list of supported encodings.

So while ultimately it's the same code, this seems like it's adding unnecessary maintenance overhead should this change in a future python version.

Reply:
Ok, that works

Reply:
That's set in the latest commit. If there are any issues though, by all means please let me know and I'll address that as soon as possible for you 👍

Thanks for your time reviewing and your consideration.